src/OSPSuite.UI/Services/DataImporter.cs

```diff
@@ -169,5 +177,45 @@ DataImporterSettings dataImporterSettings
             return result;
          }
       }
+
+      public Cache<string, IEnumerable<DataRepository>> ReloadFromConfiguration(IEnumerable<DataRepository> dataSetsToImport,
```

Comment:
Why don't you return a triple with 3  names parameters instead of relying on a cache with keys?

Reply:
To be honest that was the original way I implemented this, but I found the solution with th Cache a bit more readable. As far as I understand your argument is about stability right? I could go back to having the triple...

Reply:
done